tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -0,0 +1,155 @@
+  priority: boolean;
+  round_duration: string;
+};
+
+// Specifically not typed for flexibility.
+export const JobEntry: SFC<{ job; department; act: Function }> = (data) => {
+  const job = data.job;
+  const department = data.department;
+  return (
+    <Stack.Item fill>
+      <Button
+        width="100%"
+        style={{
+          // Try not to think too hard about this one.
+          'background-color': job.unavailable_reason
+            ? 'lightgrey'
+            : job.prioritized
+              ? '#308d25'
+              : Color.fromHex(department.color)
+                .darken(10)
+                .toString(),
+          'color': job.unavailable_reason
+            ? 'dimgrey'
+            : Color.fromHex(department.color)
+              .darken(90)
+              .toString(),
+          'font-size': '1.1rem',
+          'cursor': job.unavailable_reason ? 'initial' : 'pointer',
+        }}
+        tooltip={
+          job.unavailable_reason ? (
+            job.unavailable_reason
+          ) : job.prioritized ? (
+            <div>
+              <b>The HoP wants more people in this job!</b>
+              <br /> <br />
+              {job.description}
+            </div>
+          ) : (
+            job.description
+          )
+        }
+        onClick={() => {
+          !job.unavailable_reason && data.act('SelectedJob', { job: job.name });
+        }}
+        content={
+          <div>
+            {(job.icon = job.icon || JobToIcon[job.name] || null) && (
+              <Icon name={job.icon} />
+            )}
+            {job.command ? <b>{job.name}</b> : job.name}
+            <span style={{ 'float': 'right' }}>
+              {job.used_slots} / {job.open_slots}
+            </span>
+          </div>
+        }
+      />
+    </Stack.Item>
+  );
+};
+
+export const JobSelection = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { departments } = data;
+
+  let any_job_available = false;
+
+  departments.forEach((department) => {
+    department.jobs.forEach((job) => {
+      if (!any_job_available && !job.unavailable_reason) {
+        any_job_available = true;
+      }
+    });
+  });
+  return (
+    <Window width={1012} height={716}>
+      <Window.Content scrollable>
+        <DepartmentEntry
+          title="Job Selection"
+          titleContents={
+            <Button
+              style={{ 'float': 'right' }}
+              onClick={() => act('SelectedJob', { 'job': 'Random' })}
+              content="Random Job!"
+              tooltip="Roll a random job. You can re-roll or cancel your random job if you don't like it."
+            />
+          }>
+          <DepartmentPane
+            act={act}
+            departments={departments}
+            titleSubtextBuilder={(department) => {
+              return (
+                department['open_slots'] && (
+                  <span
+                    style={{
+                      'float': 'right',
```

Comment:
Same here for if we shouldn't be using float

Reply:
Ditto